Source/MythicaEditor/Private/MythicaEditorSubsystem.cpp

```diff
@@ -1362,10 +1385,27 @@ void UMythicaEditorSubsystem::OnExecuteJobResponse(FHttpRequestPtr Request, FHtt
     SetJobState(RequestId, EMythicaJobState::Queued);
 }
 
-int UMythicaEditorSubsystem::CreateJob(const FString& JobDefId, const FMythicaParameters& Params, const FString& ImportPath)
+int UMythicaEditorSubsystem::CreateJob(const FString& JobDefId, const FMythicaParameters& Params, const FString& ImportPath, UMythicaComponent* Component)
 {
     int RequestId = NextRequestId++;
-    Jobs.Add(RequestId, { JobDefId, {}, Params, ImportPath });
+
+    FMythicaJob& Job = Jobs.Add(RequestId, { JobDefId, {}, Params, ImportPath });
+
+    Job.StartTime = FDateTime::Now();
+
+    if (IsValid(Component))
+    {
+        FString DisplayName = UKismetSystemLibrary::GetDisplayName(Component);
+        FMythicaRequestIdList& RequestList = ComponentToJobs.FindOrAdd(DisplayName);
+        RequestList.RequestIds.EmplaceAt(0, RequestId);
+
+        OnJobCreated.Broadcast(RequestId, DisplayName);
+    }
+    else
+    {
+        // @TODO - Liam: Properly handle this case were we dont have a valid component
+        OnJobCreated.Broadcast(RequestId, FString());
```

Comment:
Can this case happen? Another option could be to make the function take it in as a reference if this case isn't possible.